build.gradle

```diff
@@ -429,3 +451,14 @@ tasks.register('downloads') {
 	}
 	inputs.files neededDownloaders
 }
+
+////////////////////////////////////////////////////////////////////////
+//
+//  Helpers for dependency locking
+//
+
+// this task resolves dependencies in all sub-projects, making it easy to
+// generate lockfiles
+allprojects {
+    task allDeps(type: DependencyReportTask) {}
```

Comment:
Use [the task configuration avoidance APIs](https://docs.gradle.org/current/userguide/task_configuration_avoidance.html) here:

```groovy
tasks.register('allDeps', DependencyReportTask) {}
```

Or can we remove this task entirely? According to [the `DependencyReportTask` documentation](https://docs.gradle.org/current/dsl/org.gradle.api.tasks.diagnostics.DependencyReportTask.html),

> An instance of this type is used when you execute the `dependencies` task from the command-line.

If that's the case, then why do we need an extra `allDeps` task?

Reply:
I think the key here is that we are adding the task to `allprojects`, which includes all sub-projects.  When I run `./gradlew dependencies` at the top level, I get:
```
$ ./gradlew dependencies
Starting a Gradle Daemon (subsequent builds will be faster)

> Task :dependencies

------------------------------------------------------------
Root project
------------------------------------------------------------

No configurations
```

I added the `allDeps` task as a way to run the `dependencies` task on all sub-projects; just running `./gradlew allDeps` does the trick.  Do you know of a different way to do this that doesn't require adding another task?  In Googling around I didn't see anything other than equivalents of the `allDeps` task.

Reply:
> Use [the task configuration avoidance APIs](https://docs.gradle.org/current/userguide/task_configuration_avoidance.html) here:

4a0a78a05fabcef9eaee6b991886e13d0d90d480